### PR TITLE
Support Indonesia names, address, phone & gender

### PR DESCRIPTION
**Pull request by brain64bit**
Add support data for Indonesia like male & female 
names, states & address, genders and also both
home & mobile phone number for Indonesia.

Some enhancements are:
- FFaker::NameID
- FFaker::AddressID
- FFaker::PhoneNumberID
- FFaker::GenderID

### CODE REVIEW

File: test/test_address_id.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  def test_state
+    assert FFaker::AddressID::STATE.include?(FFaker::AddressID.state)
+  end
+
+  def test_state_abbr
+    assert FFaker::AddressID::STATE_ABBR.include?(FFaker::AddressID.state_abbr)
+  end
+
+  def test_city
+    assert FFaker::AddressID::CITY.include?(FFaker::AddressID.city)
```
Comment (marocchino) [inline test/test_address_id.rb]: question) How do you distinguish between assert_include and assert?
If you do not have a good reason, How about unify with assert_include?
```
assert FFaker::AddressID::CITY.include?(FFaker::AddressID.city)
assert_include(FFaker::AddressID::CITY, FFaker::AddressID.city)
```

File: test/test_name_id.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def test_name
+    assert_match(/\A[a-zA-Z\s]+\z/, @tester.name)
+  end
+
+  def test_name_with_prefix
+    prefix, name, last_name = @tester.name_with_prefix.split(/\s+/)
+    assert_include(@tester::PREFIXES, prefix)
+    assert(!name.empty?)
```
Comment (marocchino) [inline test/test_name_id.rb]: You can use refute_empty here.

File: test/test_phone_number_id.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    :phone_number
+  )
+
+  def setup
+    @tester = FFaker::PhoneNumberID
+  end
+
+  def test_mobile_phone_number
+    prefixes = FFaker::PhoneNumberID::MOBILE_PHONE_PREFIXES
+    regex = /(#{prefixes.join('|')})-(\d{8})/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: https://github.com/bbatsov/ruby-style-guide#non-capturing-regexp
`(\d{8})` capture parts are not used and can be removed.
And the matching of this file does not check the entire string at all.

`/\A(?:#{prefixes.join('|')})-\d{8}\z/`

File: lib/ffaker/phone_number_id.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    HOME_PHONE_PREFIXES = HOME_PHONE_PREFIXES.freeze
+
+    def mobile_phone_number
+      "#{fetch_sample(MOBILE_PHONE_PREFIXES)}-#{FFaker.numerify('########')}"
+    end
+
+    def home_phone_number
+      case rand(2)
+      when 0 then "#{fetch_sample(HOME_PHONE_PREFIXES)}-#{FFaker.numerify('#######')}"
+      when 1 then "#{fetch_sample(HOME_PHONE_PREFIXES)}-#{FFaker.numerify('########')}"
```
Comment (marocchino) [inline lib/ffaker/phone_number_id.rb]: There is a repetition of code.
How about repeat it in a narrower range?

```rb
rest_nums = ....
"#{fetch_sample(HOME_PHONE_PREFIXES)}-#{FFaker.numerify(rest_nums)}"
```

File: lib/ffaker/address_id.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+module FFaker
+  module AddressID
+    include FFaker::Address
+
+    extend ModuleUtils
+    extend self
+
+    STREET_PREFIX = %w(Jl Jln).freeze
+    COMMON_STREET_NAMES = COMMON_STREET_NAMES.freeze
```
Comment (marocchino) [inline lib/ffaker/address_id.rb]: Get rid of this line. :3

File: test/test_address_id.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def test_street_prefix
+    assert FFaker::AddressID::STREET_PREFIX.include?(FFaker::AddressID.street_prefix)
+  end
+
+  def test_street
+    prefixes = FFaker::AddressID::STREET_PREFIX
+    common_street_names = FFaker::AddressID::COMMON_STREET_NAMES
+    regex = /(#{prefixes.join('|')})\.(\s(#{common_street_names.join('|')}))\,\sNo\.\s\d+/
```
Comment (marocchino) [inline test/test_address_id.rb]: It does not test whole string.

File: test/test_phone_number_id.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  def test_mobile_phone_number
+    prefixes = FFaker::PhoneNumberID::MOBILE_PHONE_PREFIXES
+    regex = /\A(?:#{prefixes.join('|')})-\d{8}\z/
+    assert_match regex, FFaker::PhoneNumberID.mobile_phone_number
+  end
+
+  def test_home_phone_number
+    prefixes = FFaker::PhoneNumberID::HOME_PHONE_PREFIXES
+    regex = /(#{prefixes.join('|')})-(\d{7,8})$/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: It does not test whole string.

File: test/test_phone_number_id.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    assert_match regex, FFaker::PhoneNumberID.home_phone_number
+  end
+
+  def test_phone_number
+    mobile_prefixes = FFaker::PhoneNumberID::MOBILE_PHONE_PREFIXES
+    home_prefixes = FFaker::PhoneNumberID::HOME_PHONE_PREFIXES
+    regex = /(#{mobile_prefixes.join('|')}|#{home_prefixes.join('|')})-(\d{7,8})/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: It does not test whole string.

File: test/test_phone_number_id.rb [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    assert_match regex, FFaker::PhoneNumberID.phone_number
+  end
+
+  def test_international_mobile_phone_number
+    prefixes = FFaker::PhoneNumberID::MOBILE_PHONE_PREFIXES.map { |x| x.sub('0', '') }
+    regex = /\+62(#{prefixes.join('|')})-(\d{8})/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: It does not test whole string.

File: test/test_phone_number_id.rb [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    assert_match regex, FFaker::PhoneNumberID.international_mobile_phone_number
+  end
+
+  def test_international_home_phone_number
+    prefixes = FFaker::PhoneNumberID::HOME_PHONE_PREFIXES.map { |x| x.sub('0', '') }
+    regex = /\+62(#{prefixes.join('|')})-(\d{7,8})/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: It does not test whole string.

File: test/test_phone_number_id.rb [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    assert_match regex, FFaker::PhoneNumberID.international_home_phone_number
+  end
+
+  def test_international_phone_number
+    mobile_prefixes = FFaker::PhoneNumberID::MOBILE_PHONE_PREFIXES.map { |x| x.sub('0', '') }
+    home_prefixes = FFaker::PhoneNumberID::HOME_PHONE_PREFIXES.map { |x| x.sub('0', '') }
+    regex = /\+62(#{mobile_prefixes.join('|')}|#{home_prefixes.join('|')})-(\d{7,8})/
```
Comment (marocchino) [inline test/test_phone_number_id.rb]: It does not test whole string.

Comment (brain64bit) [inline test/test_phone_number_id.rb]: @marocchino thanks for review, i'm already fix your issues

File: lib/ffaker/phone_number_id.rb [hunk 12]
```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
```
Comment (b264) [inline lib/ffaker/phone_number_id.rb]: It needs `# frozen_string_literal: true` as the first line

File: test/test_phone_number_id.rb [hunk 13]
```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
```
Comment (b264) [inline test/test_phone_number_id.rb]: It needs `# frozen_string_literal: true` as the first line in the file

**Comment by marocchino** [comment]
LGTM

**Pull request merged**